tests/udf_communication/peer_communication/test_add_peer.py

```diff
@@ -0,0 +1,86 @@
+import time
+import traceback
+from pathlib import Path
+from typing import Dict, List
+
+import structlog
+from structlog import WriteLoggerFactory
+from structlog.types import FilteringBoundLogger
+
+from exasol_advanced_analytics_framework.udf_communication.connection_info import ConnectionInfo
+from exasol_advanced_analytics_framework.udf_communication.ip_address import IPAddress
+from exasol_advanced_analytics_framework.udf_communication.peer import Peer
+from exasol_advanced_analytics_framework.udf_communication.peer_communicator import key_for_peer, PeerCommunicator
+from tests.udf_communication.peer_communication.utils import TestProcess, BidirectionalQueue, assert_processes_finish
+
+structlog.configure(
+    context_class=dict,
+    logger_factory=WriteLoggerFactory(file=Path(__file__).with_suffix(".log").open("wt")),
+    processors=[
+        structlog.contextvars.merge_contextvars,
+        structlog.processors.add_log_level,
+        structlog.processors.StackInfoRenderer(),
+        structlog.dev.set_exc_info,
+        structlog.processors.TimeStamper(),
+        structlog.processors.JSONRenderer()
+    ]
+)
+
+LOGGER: FilteringBoundLogger = structlog.get_logger(__name__)
+
+
+def run(name: str, group_identifier: str, number_of_instances: int, queue: BidirectionalQueue):
+    logger = LOGGER.bind(group_identifier=group_identifier, name=name)
+    try:
+        listen_ip = IPAddress(ip_address=f"127.1.0.1")
+        com = PeerCommunicator(
+            name=name,
+            number_of_peers=number_of_instances,
+            listen_ip=listen_ip,
+            group_identifier=group_identifier)
+        try:
+            queue.put(com.my_connection_info)
+            peer_connection_infos = queue.get()
+            for index, connection_info in peer_connection_infos.items():
+                com.register_peer(connection_info)
+            peers = com.peers(timeout_in_milliseconds=None)
+            logger.info("peers", peers=peers)
+            queue.put(peers)
+        finally:
+            com.close()
+    except Exception as e:
+        traceback.print_exc()
+        logger.exception("Exception during test", exception=e)
+
+
+def test():
+    group = f"{time.monotonic_ns()}"
+    logger = LOGGER.bind(group=group, location="test")
+    logger.info("start")
+    number_of_instances = 10
```

Comment:
Does it make sense to use parametrized test here, maybe with 1, 10 and 100 instances?

Reply:
see my last changes